pkg/operator/status/status_controller.go

```diff
@@ -0,0 +1,407 @@
+		}
+	}
+	return conditions
+}
+
+func computeClusterOperatorVersions() []configv1.OperandVersion {
+	currentVersion := os.Getenv("RELEASE_VERSION")
+	version := []configv1.OperandVersion{
+		{
+			Name:    "operator",
+			Version: currentVersion,
+		},
+	}
+	return version
+}
+
+func setLastTransitionTime(oldConditions []configv1.ClusterOperatorStatusCondition, newConditions []configv1.ClusterOperatorStatusCondition) {
+	for i := range newConditions {
+		newCondition := &newConditions[i]
+		oldCondition, _ := findClusterOperatorCondition(oldConditions, newCondition.Type)
+		if oldCondition == nil || !ConditionEqual(*oldCondition, *newCondition) {
+			newCondition.LastTransitionTime = metav1.Now()
+		} else {
+			newCondition.LastTransitionTime = oldCondition.LastTransitionTime
+		}
+	}
+}
+
+// ConditionEqual compares every field except LastTransitionTime.
+func ConditionEqual(a, b configv1.ClusterOperatorStatusCondition) bool {
+	if a.Type == b.Type &&
+		a.Status == b.Status &&
+		a.Reason == b.Reason &&
+		a.Message == b.Message {
+		return true
+	}
+	return false
+}
+
+// isWatchedSecret is used to identify if a given secret namespace + name is one we're expecting in the future
+// or the root credential for the cluster, or a CCO-managed secret.
+func isWatchedSecret(platformType configv1.PlatformType, namespace, name string, annotations map[string]string) bool {
+	for _, nsn := range getWatchedSecrets(platformType) {
+		if nsn.Namespace == namespace && nsn.Name == name {
+			return true
+		}
+	}
+
+	if _, ok := annotations[credreqv1.AnnotationCredentialsRequest]; ok {
+		return true
+	}
+
+	return false
+}
+
+func getWatchedSecrets(platformType configv1.PlatformType) []types.NamespacedName {
+	secrets := []types.NamespacedName{}
+
+	rootSecret := types.NamespacedName{
+		Namespace: constants.CloudCredSecretNamespace,
+		// Fill in platform-specific name below
+	}
+
+	switch platformType {
+	case configv1.AWSPlatformType:
+		rootSecret.Name = constants.AWSCloudCredSecretName
+		secrets = append(secrets, constants.AWSUpcomingSecrets...)
+	case configv1.AzurePlatformType:
+		rootSecret.Name = constants.AzureCloudCredSecretName
+		secrets = append(secrets, constants.AzureUpcomingSecrets...)
+	case configv1.GCPPlatformType:
+		rootSecret.Name = constants.GCPCloudCredSecretName
+		secrets = append(secrets, constants.GCPUpcomingSecrets...)
+	case configv1.OpenStackPlatformType:
+		rootSecret.Name = constants.OpenStackCloudCredsSecretName
+		secrets = append(secrets, constants.OpenStackUpcomingSecrets...)
+	case configv1.OvirtPlatformType:
+		rootSecret.Name = constants.OvirtCloudCredsSecretName
+		secrets = append(secrets, constants.OvirtUpcomingSecrets...)
+	case configv1.VSpherePlatformType:
+		rootSecret.Name = constants.VSphereCloudCredSecretName
+		secrets = append(secrets, constants.VsphereUpcomingSecrets...)
+	default:
+		log.Errorf("unable to provide upcoming secrets for unknown platform: %v", platformType)
```

Comment:
This looks like it will be called on bare metal, so hitting this line is really not an error IMO, I'd go with info (or maybe even debug) and soften the message up a little bit.